eks_infrastructure/upgrade_operation.sh

```diff
@@ -56,11 +56,16 @@ function upgrade_nodegroups() {
         --cluster ${CLUSTER} \
         --kubernetes-version ${EKS_VERSION} \
         --timeout 90m \
-        --region ${REGION}
+        --region ${REGION} || UPGRADE_FAILED=1
```

Comment:
Can you instead define a bash array so we can keep track of the nodegroups that failed, and if the array is not empty, print out the failed nodegroups?